allennlp/nn/decoding/decoder_trainers/expected_risk_minimization.py

```diff
@@ -49,7 +49,11 @@ def decode(self,
             # Finished model scores are log-probabilities of the predicted sequences. We convert
             # log probabilities into probabilities and re-normalize them to compute expected cost under
             # the distribution approximated by the beam search.
-            costs = torch.cat(finished_costs[batch_index])
+
+            # finished_costs[batch_index] could be a list of 0-tensors or 1-tensors. We use .view(-1)
+            # to make sure they're treated as 1-tensors and then concatenated appropriately.
+            # TODO(joelgrus): make sure this is correct
```

Comment:
This looks like it shouldn't be right to me.  Well, it looks like it probably does the right thing, but I think fixing this where the `finished_costs` gets created is the right solution, instead of this one.  My guess is that something is getting created as a scalar now, where we really want it as a 1-dim tensor.